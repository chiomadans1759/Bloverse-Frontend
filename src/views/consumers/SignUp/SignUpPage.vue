<template>
  <div class="sign-up-container">
    <div class="top">
      <div>
        <span class="top-left-container">
        <a href="/login">
          <i class="fal fa-long-arrow-left fa-2x"></i>
          <span class="back-text">Back</span>
        </a>
        <router-link to="/" class="router-link">
          <img class="logo" src="@/assets/Bloverse-white.svg">
          <img class="blue-logo" src="@/assets/Bloverse.svg">
        </router-link>
      </span>
      </div>
    </div>
    <div class="sign-up-content d-flex flex-column justify-content-center align-items-center">
      <div>
        <h2 class="sign-up-intro-one">Join Us</h2>
        <h4 class="sign-up-intro-two">Lorem ipsum dolor sit amet, vel
          <br /> accumsanliberse ex, ea nec
        </h4>
        <h5 class="sign-up-intro-three">Sign Up With</h5>

        <div class="social-login">
          <Button customClass="btn btn-light sign-up-social-button sign-up-facebook-btn">
            <img src="@/assets/facebook-logo-light.svg">
          </Button>
          <br>
          <Button customClass="btn btn-light sign-up-social-button">
            <img src="@/assets/google-text-logo.png" height="25" width="60">
          </Button>
        </div>

        <div class="grid-container">
          <div class="grid-item">
            <hr class="hr-text" />
          </div>
          <div class="middle-grid-item">
            <p><strong>Or</strong></p>
          </div>
          <div class="grid-item">
            <hr class="hr-text" />
          </div>
      </div>
    </div>

    <div class="sign-up-form-section">
      <text-input
        id="first-name"
        name="firstName"
        type="text"
        placeholder="First name"
        :value="firstName"
        :onChange="onChange"
      />
      <text-input
        id="last-name"
        name="lastName"
        type="text"
        placeholder="Last name"
        :value="lastName"
        :onChange="onChange"
      />
      <text-input
        id="email"
        name="email"
        type="email"
        placeholder="Email"
        :value="email"
        :onChange="onChange"
      />
      <text-input
        id="password"
        name="password"
        type="password"
        placeholder="Password"
        :value="password"
        :onChange="onChange"
      />
      <Button customClass="btn btn-primary primary-button">Sign Up</Button>

      <h4 class="sign-up-terms-and-privacy">
        By creating an account, you agree to our
        <router-link to="/terms-and-conditions" class="sign-up-link-text">Terms</router-link>
        and
        <router-link to="/privacy-policies" class="sign-up-link-text">Privacy</router-link>
      </h4>

      <Button customClass="btn btn-outline-secondary secondary-button">
        <h2 class="sign-up-new-bloverse-text">Have an account?
          <router-link to="/login" class="signup-login-text">Login</router-link>
        </h2>
      </Button>
    </div>

      <div class="signup-creator-text">
        <h2 class="sign-up-creator-text">Content creator? 
        <router-link to="/creators/apply" class="router-link sign-up-apply-now-text">
          Apply
        </router-link>
        </h2>
      </div>
    </div>
    <creator-card />
    <div class="bottom">
      <p><span class="bloverse">Bloverse</span> {{year}}. All Right reserved</p>
    </div>
  </div>
</template>

<script>
import Button from "../../../components/Button/Button.vue";
import TextInput from "../../../components/TextInput/TextInput.vue";
import CreatorCard from "../../../components/CreatorCard/CreatorCard.vue";

export default {
  data() {
    return {
      firstName: "",
      lastName: "",
      email: "",
      password: "",
      year: new Date().getFullYear()
    };
  },
  components: {
    Button,
    TextInput,
    CreatorCard,
  },
  methods: {
    onChange: function(event) {
      this[event.target.name] = event.target.value;
    }
  },
  mounted() {
    /* eslint-disable */
    hideTawkBot();
    window.onload = function() {
      hideTawkBot();
    };
  },
  destroyed() {
    showTawkBot();
  }
};
</script>

<style lang="scss">
@import "./SignUpPage.scss";
</style>
