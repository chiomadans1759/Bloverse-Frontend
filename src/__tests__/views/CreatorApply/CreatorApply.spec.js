import { mount, RouterLinkStub } from '@vue/test-utils';
import SignUp from '@/views/journalists/CreatorApply/CreatorApplyPage.vue';

describe('Creator Apply Component', () => {
  it('should render correctly', () => {
    const wrapper = mount(SignUp, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should add a field', () => {
    const wrapper = mount(SignUp, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    wrapper.find('#add-article').trigger('click');

    expect((wrapper.vm.articleInputs).length).toBe(1);
  });

  it('should delete a field', () => {
    const wrapper = mount(SignUp, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });
    wrapper.find('#add-article').trigger('click');
    wrapper.find('.delete-field').trigger('click');

    expect((wrapper.vm.articleInputs).length).toBe(0);
  })

  it('is called when youngVoice value changes', () => {
    const wrapper = mount(SignUp, {
      stubs: {
        RouterLink: RouterLinkStub
      }
    });

    wrapper.vm.youngVoice = 'yes';
    expect(wrapper.vm.isYoungVoice).toBe(true);

    wrapper.vm.youngVoice = 'no';
    expect(wrapper.vm.isYoungVoice).toBe(false);
  })
});
