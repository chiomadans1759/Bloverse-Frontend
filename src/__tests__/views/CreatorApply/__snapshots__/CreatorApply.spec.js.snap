// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creator Apply Component should render correctly 1`] = `
<div
  class="container-fluid creator-apply"
>
  <div
    class="row p-3 ml-4"
  >
    <span
      class="top-left-container d-flex justify-content-center align-items-center"
    >
      <a
        class="d-flex justify-content-around"
        href="/"
      >
        <i
          class="fal fa-long-arrow-left fa-2x"
        />
         
        <span
          class="back-text"
        >
          Back
        </span>
      </a>
       
      <a
        class="router-link pl-4"
      >
        <img
          class="nav-logo"
          src="@/assets/Logo.svg"
        />
      </a>
    </span>
  </div>
   
  <div
    class="row py-5 d-flex"
  >
    <div
      class="offset-lg-4 col-lg-3 offset-md-3 col-md-6 col-sm-12"
    >
      <div
        class="col-12"
      >
        <h1
          class="creator-apply__title"
        >
          Apply
        </h1>
         
        <p
          class="creator-apply__sub-title pt-2"
        >
          Join Bloverse as acontent creator
        </p>
      </div>
       
      <div
        class="col-12 pt-4"
      >
        <div
          class="form-group"
        >
          <input
            aria-describedby="firstName"
            class="form-control"
            placeholder="First Name*"
            type="text"
          />
        </div>
         
        <div
          class="form-group"
        >
          <input
            aria-describedby="lastName"
            class="form-control"
            placeholder="Last Name*"
            type="text"
          />
        </div>
         
        <div
          class="form-group"
        >
          <input
            aria-describedby="emailHelp"
            class="form-control"
            placeholder="Email"
            type="email"
          />
        </div>
         
        <div
          class="form-group"
        >
          <div
            class="input-group mb-2 mr-sm-2"
          >
            <div
              class="input-group-prepend"
            >
              <div
                class="input-group-text"
              >
                +234
              </div>
            </div>
             
            <input
              class="form-control"
              id="inlineFormInputGroupUsername2"
              placeholder="000 000 0000"
              type="number"
            />
          </div>
        </div>
         
        <div
          class="form-group"
        >
          <select
            class="custom-select"
          >
            <option
              selected="selected"
            >
              Lagos, Nigeria*
            </option>
             
            <option
              value="1"
            >
              Mexico
            </option>
             
            <option
              value="2"
            >
              UK
            </option>
             
            <option
              value="3"
            >
              Ghana
            </option>
          </select>
        </div>
         
        <div
          class="form-group"
        >
          <label
            for=""
          >
            Bloverse Young Voices?
          </label>
           
          <br />
           
          <div
            class="custom-control custom-radio custom-control-inline"
          >
            <input
              class="custom-control-input"
              id="customRadioInline2"
              name="customRadioInline1"
              type="radio"
              value="yes"
            />
             
            <label
              class="custom-control-label pt-1"
              for="customRadioInline2"
            >
              Yes
            </label>
          </div>
           
          <div
            class="custom-control custom-radio custom-control-inline"
          >
            <input
              class="custom-control-input"
              id="customRadioInline1"
              name="customRadioInline1"
              type="radio"
              value="no"
            />
             
            <label
              class="custom-control-label pt-1"
              for="customRadioInline1"
            >
              No
            </label>
          </div>
        </div>
         
        <div
          class="form-group"
        >
          <select
            class="custom-select"
            style="display: none;"
          >
            <option
              selected="selected"
            >
              Select your institution*
            </option>
             
            <option
              value="1"
            >
              Mexico
            </option>
             
            <option
              value="2"
            >
              UK
            </option>
             
            <option
              value="3"
            >
              Ghana
            </option>
          </select>
        </div>
         
        <div
          class="form-group pb-5"
        >
          <label
            class="d-flex flex-column"
            for=""
          >
            <strong>
              Category
            </strong>
             
            <span>
              Ipsu m lorem torem di
            </span>
          </label>
           
          <select
            class="custom-select"
          >
            <option
              selected="selected"
            >
              Choose category*
            </option>
             
            <option
              value="1"
            >
              Mexico
            </option>
             
            <option
              value="2"
            >
              UK
            </option>
             
            <option
              value="3"
            >
              Ghana
            </option>
          </select>
           
          <ul />
           
          <span
            class="float-right pt-1 link"
          >
            Add More Category
          </span>
        </div>
         
        <div>
          <div
            class="form-group"
          >
            <label
              class="d-flex flex-column"
              for=""
            >
              <strong>
                Links
              </strong>
               
              <span>
                Ipsu m lorem torem di
              </span>
            </label>
          </div>
           
          <div
            class="form-group"
          >
            <input
              aria-describedby="emailHelp"
              class="form-control"
              id="exampleInputEmail1"
              placeholder="Instagram Link"
              type="text"
            />
          </div>
           
          <div
            class="form-group"
          >
            <input
              aria-describedby="emailHelp"
              class="form-control"
              id="exampleInputEmail1"
              placeholder="LinkedIn Link"
              type="text"
            />
          </div>
           
          <div
            class="form-group"
          >
            <input
              aria-describedby="emailHelp"
              class="form-control"
              id="exampleInputEmail1"
              placeholder="Twitter Link"
              type="text"
            />
          </div>
           
          <div>
            <input
              aria-describedby="emailHelp"
              class="form-control"
              id="exampleInputEmail1"
              placeholder="Link to written article"
              type="text"
            />
             
            <ul />
             
            <span
              class="float-right pt-2 pb-3 link"
              id="add-article"
            >
              Add More link To Written Article
            </span>
          </div>
        </div>
         
        <div
          class="form-group"
        >
          <div
            data-toggle="modal"
            href="#success-modal"
          >
            <button
              class="btn btn-primary btn-block edit-btn mb-4"
            >
              Apply
            </button>
          </div>
           
          <label
            class="pb-4"
          >
            By creating an account, you agree to our
            
            <a
              class="sign-up-link-text"
            >
              Terms
            </a>
            and
            
            <a
              class="sign-up-link-text"
            >
              Privacy
            </a>
          </label>
           
          <div
            class="border p-3 bg-white creator-apply__login"
          >
            <span>
              Have an account?
              
              <a
                class="signup-login-text"
              >
                Login
              </a>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <main
    id="bl-modal"
  >
    <div
      aria-hidden="true"
      aria-labelledby="customModal"
      class="modal fade"
      id="success-modal"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-dialog-centered"
        role="document"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header darkBackground"
          >
            <h5
              class="modal-title"
              id="customModal"
            >
              Your application has been sent to bloverse.
            </h5>
             
            <div>
              <!---->
               
              <!---->
            </div>
             
            <button
              aria-label="Close"
              class="close"
              data-dismiss="modal"
              type="button"
            >
              <span
                aria-hidden="true"
              >
                Ã—
              </span>
            </button>
          </div>
           
          <div
            class="modal-body"
          >
            <div
              class="modal-body"
            >
              <p>
                A message will be sent to your mail to continue the verification and approval process in 48hrs.
              </p>
            </div>
          </div>
           
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-outline-primary"
              data-dismiss="modal"
              type="button"
            >
              Close
            </button>
             
            <button
              class="btn btn-primary"
              type="button"
            >
              Return
            </button>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
